ravendb/rvn_server.go

```diff
@@ -0,0 +1,824 @@
+	}
+
+	if cert, ok := ns.Assets["certificate.pfx"]; ok {
+		ns.ClusterCertificate = cert
+		delete(ns.Assets, "certificate.pfx")
+	}
+
+	store, err := getStore(sc, index)
+	if err != nil {
+		return ns, err
+	}
+	buildNumber := operations.OperationGetBuildNumber{}
+	err = executeWithRetries(store, &buildNumber)
+	if err != nil {
+		return ns, err
+	}
+
+	ns.Version = strconv.Itoa(buildNumber.BuildVersion)
+
+	ns.Host = publicIP
+	ns.TcpUrl = ns.Settings["PublicServerUrl"].(string)
+	ns.HttpUrl = ns.Settings["PublicServerUrl.Tcp"].(string)
+	if unsecuredAccessAllowed, ok := ns.Settings["Security.UnsecuredAccessAllowed"]; ok {
+		ns.Insecure = unsecuredAccessAllowed == "PublicNetwork"
+	}
+
+	delete(ns.Settings, "PublicServerUrl")
+	delete(ns.Settings, "PublicServerUrl.Tcp")
+	delete(ns.Settings, "Security.UnsecuredAccessAllowed")
+
+	return ns, nil
+}
+
+func (sc *ServerConfig) deployServer(publicIP string, index int) (err error) {
+	var stdoutBuf bytes.Buffer
+	var conn *ssh.Client
+	defer func() {
+		log.Println(stdoutBuf.String())
+	}()
+	ravenPackageUrl := "https://daily-builds.s3.us-east-1.amazonaws.com/ravendb_" + sc.Package.Version + sc.Package.Arch
+
+	signer, err := ssh.ParsePrivateKey(sc.SSH.Pem)
+	if err != nil {
+		return err
+	}
+	authConfig := &ssh.ClientConfig{
+		User:            sc.SSH.User,
+		Auth:            []ssh.AuthMethod{ssh.PublicKeys(signer)},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+		Timeout:         1 * time.Minute,
+	}
+	hostAndPort := net.JoinHostPort(publicIP, fmt.Sprint(sc.SSH.getPort()))
+	for i := 0; i <= numberOfIterationsExecuteWithRetries; i++ {
+		conn, err = ssh.Dial("tcp", hostAndPort, authConfig)
+		if err != nil && i < numberOfIterationsExecuteWithRetries {
+			time.Sleep(time.Second * 2)
+		} else if err == nil {
+			break
+		} else {
+			return errors.New("Unable to SSH to " + hostAndPort + " because " + err.Error())
+		}
+	}
+	defer conn.Close()
+	err = sc.execute(publicIP, []string{
+		"n=0; while [ \"$n\" -lt 10 ] && [ ! -f /var/lib/cloud/instance/boot-finished ]; do echo 'Waiting for cloud-init...'; n=$(( n + 1 )); sleep 1; done",
+		"wget -nv -O ravendb.deb " + ravenPackageUrl,
+		"while ! sudo apt-get update -y; do sleep 1; done", //sometimes 'sudo apt get-update' fails.
+		"sudo apt-get install -y -f ./ravendb.deb",
+	}, "", &stdoutBuf, conn)
+	if err != nil {
+		return err
+	}
+
+	err = upload(conn, stdoutBuf, "/etc/ravendb/license.json", sc.License)
+	if err != nil {
+		return err
+	}
+
+	contents, err := readFileContents("/etc/ravendb/settings.json", stdoutBuf, conn)
+	if err != nil {
+		return err
+	}
+
+	var settings map[string]interface{}
+	err = json.Unmarshal(contents, &settings)
+	if err != nil {
+		stdoutBuf.WriteString("Failed to ravendb settings.json\n")
+		stdoutBuf.Write(contents)
+		return err
+	}
+
+	////stub
+	//for path, content := range sc.Assets {
+	//	err = upload(conn, stdoutBuf, path, content)
+	//	if err != nil {
+	//		return err
+	//	}
+	//}
```

Comment:
Remove this

Reply:
Where are we actually uploading the assests?

Reply:
We don't yet. We'll do that later.